### PR TITLE
display features.

### PR DESCRIPTION
**Pull request by Alice1319**
fixes eclipse-xpanse/xpanse#1547
fixes eclipse-xpanse/xpanse#1536

### CODE REVIEW

File: src/components/content/order/common/FlavorInfo.tsx [hunk 1]
```diff
@@ -3,44 +3,105 @@
...
-    disabled,
     onChangeFlavor,
 }: {
     selectFlavor: string;
-    flavorList?: Flavor[];
-    disabled?: boolean;
+    flavorList?: ServiceFlavor[];
     onChangeFlavor?: (newFlavor: string) => void;
 }): React.JSX.Element => {
+    const [maxFeatures, setMaxFeatures] = useState<number>(0);
```
Comment (swaroopar) [inline src/components/content/order/common/FlavorInfo.tsx:21]: @Alice1319 why do we need this?

File: src/components/content/order/common/FlavorInfo.tsx [hunk 2]
```diff
@@ -3,44 +3,105 @@
...
+                                        <Radio.Button
+                                            key={flavor.name}
+                                            value={flavor.name}
+                                            style={{ height: `${String(maxFeatures)}px` }}
+                                        >
+                                            {getFlavorTitle(flavor.name)}
+                                            <div className={flavorStyles.flavorPriceContent}>
+                                                <Tag color={'blue'}>
+                                                    {/* TODO Will be fixed after #1597 is fixed */}
+                                                    {(20).toString().concat(' ').concat('CNY/').concat('hourly')}
```
Comment (swaroopar) [inline src/components/content/order/common/FlavorInfo.tsx:88]: Please keep currency "EUR" as the default

File: src/components/content/order/scale/Scale.tsx [hunk 3]
```diff
@@ -122,6 +126,40 @@ export const Scale = ({
         setScaleWarning('');
     };
 
+    const getFlavorTitle = (title: string): React.JSX.Element => {
+        return <span className={'flavor-card-title'}>{title}</span>;
+    };
+
+    const getFlavorContent = (flavor: ServiceFlavor): React.JSX.Element => {
```
Comment (swaroopar) [inline src/components/content/order/scale/Scale.tsx:133]: everything here seems to be duplicate of what we have in FlavorInfo.tsx. Please remove any duplicate code.

File: src/components/content/order/scale/Scale.tsx [hunk 4]
```diff
@@ -122,6 +126,40 @@ export const Scale = ({
         setScaleWarning('');
     };
 
+    const getFlavorTitle = (title: string): React.JSX.Element => {
+        return <span className={'flavor-card-title'}>{title}</span>;
```
Comment (swaroopar) [inline src/components/content/order/scale/Scale.tsx:130]: use CssModule

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: src/components/content/order/common/FlavorInfo.tsx (hunk 1 above)
Comment (Alice1319) [reply src/components/content/order/common/FlavorInfo.tsx:21]: add a comment.

**Comment by WangLiNaruto** [comment]
LGTM

File: src/components/content/order/formDataHelpers/flavorHelper.ts [hunk 5]
```diff
@@ -23,15 +22,16 @@ export function getFlavorList(
...
-            const flavorItem = { value: flavor.name, label: flavor.name, price: (20).toString() };
-            flavors.push(flavorItem);
-        }
-    }
-
-    return flavors;
+    return flavorMapper.get(selectCsp) ?? [];
 }
+
+export const getMaxFeatureBoxSize = (flavorList: ServiceFlavor[] | undefined): number => {
```
Comment (swaroopar) [inline src/components/content/order/formDataHelpers/flavorHelper.ts:28]: we must still get rid of all this.

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Review by swaroopar — CHANGES_REQUESTED** [review]
LGTM

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**